src/ifcgeom/mapping/IfcFixedReferenceSweptAreaSolid.cpp

```diff
@@ -17,106 +17,133 @@
  *                                                                              *
  ********************************************************************************/
 
-#include "mapping.h"
 #include "../function_item_evaluator.h"
+#include "mapping.h"
 #define mapping POSTFIX_SCHEMA(mapping)
 using namespace ifcopenshell::geometry;
 
 #ifdef SCHEMA_HAS_IfcFixedReferenceSweptAreaSolid
 
+namespace {
+// Helper function to check if an IFC instance has alignment elements
+bool has_alignment_elements(const IfcSchema::IfcFixedReferenceSweptAreaSolid* inst) {
+    if (inst->Directrix()->declaration().name() == "IfcGradientCurve") {
+        return true;
+    } else {
+        return false;
+    }
+}
+} // namespace
+
 taxonomy::ptr mapping::map_impl(const IfcSchema::IfcFixedReferenceSweptAreaSolid* inst) {
-	auto dir = map(inst->Directrix());
-	auto ref = taxonomy::cast<taxonomy::direction3>(map(inst->FixedReference()));
-	auto profile = taxonomy::cast<taxonomy::face>(map(inst->SweptArea()));
-
-	auto loft = taxonomy::make<taxonomy::loft>();
-	// @todo intialize as default
-	loft->axis = nullptr;
-
-	// @todo currently only the case is handled where directrix returns a piecewise_function
-	if (auto fn = taxonomy::dcast<taxonomy::function_item>(dir)) {
-      function_item_evaluator evaluator(settings_,fn);
-		double start = 0;
-      double end = fn->length();
+    auto dir = map(inst->Directrix());
+    auto ref = taxonomy::cast<taxonomy::direction3>(map(inst->FixedReference()));
+    auto profile = taxonomy::cast<taxonomy::face>(map(inst->SweptArea()));
+
+    auto loft = taxonomy::make<taxonomy::loft>();
+    // @todo intialize as default
+    loft->axis = nullptr;
+    // Check if this instance has alignment elements
+    if (has_alignment_elements(inst)) {
+        // @todo currently only the case is handled where directrix returns a piecewise_function
+        if (auto fn = taxonomy::dcast<taxonomy::function_item>(dir)) {
```

Comment:
I don't think we need `has_alignment_elements()` actually. When the cast to a function_item fails we can assume it uses a regular IfcCurve as its directrix.

It's a bit tricky though as dcast also 'upgrades' (I think it might upgrade a polyline to a function item for example, so better use the more plain std::dynamic_pointer_cast from the stdlib. (Maybe that's what urged you to write your own function, not sure?)


```suggestion
        if (auto fn = std::dynamic_pointer_cast<taxonomy::function_item::ptr>(dir)) {
```

Reply:
Yeah, the reason I wrote a separate function was that the conditional never moved to the else condition. I'll test your suggestion and get back to you!

Reply:
using `std::dynamic_pointer_cast` seems to work! Added this in https://github.com/IfcOpenShell/IfcOpenShell/pull/7058/commits/48358297005c8955551f966bf2b632f2a07a3bd5